### DISCUSSION TITLE
Debugging is forced to terminate when stepping in after debugging has started

### DISCUSSION THREAD

**Discussion by koinumasaoko** [codefori/vscode-ibmi#2093] [Q&A] [answered]
Hi,

Debug Server and Debug Service are started without problems.
I am also able to start Debug by right-clicking on RPGLE in the IFS Browser.
However, when I step in, debug is forced to terminate, and there is no log or other output on the VSCode screen, so I am having trouble.
Is there any way to resolve the problem?
![image](https://github.com/codefori/vscode-ibmi/assets/56465596/d0e49db0-1d5b-40f2-ba52-3af13b941339)

**Environmental Information**
Code for IBM i：v2.10.4 (pre-release)
IBM i OS version：7.5 TR3

**Comment by worksofliam** [comment]
Please turn on debug tracing in the connection settings, try debugging again, and then paste the output from the debug console here - **please remove your password from the debug trace before posting!**

**Reply by koinumasaoko** [reply]
Thank you for your comment.
Sorry, some of the logs are displayed in Japanese.
CPF7E12 appears to be affected.

> To client: {"type":"response","seq":0,"request_seq":2,"success":true,"command":"launch"}
> From client: evaluate({"expression":"dcl","frameId":8,"context":"hover"})
> To client: {"type":"response","seq":0,"request_seq":12,"success":false,"command":"evaluate","body":{"error":{"id":"8999","format":"Server: CPF7E12 IDが存在しない。"}}}
> From client: stepIn({"threadId":1})
> To client: {"type":"response","seq":0,"request_seq":13,"success":true,"command":"stepIn"}
> To client: {"type":"event","seq":30,"event":"exited","body":{"exitcode":0}}
> To client: {"type":"event","seq":31,"event":"terminated"}
> From client: disconnect({"restart":false,"terminateDebuggee":false})
> To client: {"type":"response","seq":0,"request_seq":14,"success":true,"command":"disconnect"}

**Reply by koinumasaoko** [reply]
I have discovered something while debugging.
It seems that only in PGMs that reference and update the database, I cannot step in to debug.
I wonder if it has something to do with the UPDPROD parameter setting in the STRISDB command.

**Reply by koinumasaoko** [reply]
I changed the debug settings and checked the following items.

> Determines whether the job being debugged can update objects in production (*PROD) libraries.
![image](https://github.com/codefori/vscode-ibmi/assets/56465596/9e36c215-d618-457b-8ae6-9adf7ed05187)

We would also appreciate it if you could review the following logs.
```
To client: {"type":"response","seq":0,"request_seq":10,"success":true,"command":"source","body":{"content":"**free\n//Chain\nctl-opt dftname(chain);\nctl-opt dftactgrp(*no)  main(main);\n//---------------------\n// chain.rpgle\n//---------------------\n \ndcl-proc    main;\n    //parameters\n    dcl-pi *N;\n        #PTKBAN char(5);\n        #PTOKKJ char(50);\n    end-pi;\n \n    //dcl-f TOKTAB    disk    usage(*update) keyed;\n    dcl-f TOKTAB disk usage(*update) keyed rename(TOKTAB:TOKMSR);\n \n    dcl-ds  customer likerec(TOKMSR:*all);\n \n    dcl-s dsply_message varchar(52);\n \n    chain #PTKBAN TOKTAB customer;\n    if %found(TOKTAB);\n        dsply   ('B:'+ customer.TKNAKJ);\n        customer.TKNAKJ = #PTOKKJ;\n        update  TOKMSR customer;\n        dsply   ('A:'+ customer.TKNAKJ);\n    else;\n        dsply_message = 'Not found';\n        dsply   dsply_message;\n    endif;\n \nend-proc;\n"}}
To client: {"type":"event","seq":25,"event":"loadedSource","body":{"reason":"new","source":{"adapterData":{"module":"*PGM KOINULIB/CHAIN","part":"CHAIN"},"name":"/home/KOINUMA/ffrpgsrc/chain.rpgle","path":"/home/KOINUMA/ffrpgsrc/chain.rpgle","sourceReference":1001}}}
From client: stepIn({"threadId":1})
To client: {"type":"response","seq":0,"request_seq":11,"success":true,"command":"stepIn"}
To client: {"type":"event","seq":27,"event":"exited","body":{"exitcode":0}}
To client: {"type":"event","seq":28,"event":"terminated"}
From client: disconnect({"restart":false,"terminateDebuggee":false})
To client: {"type":"response","seq":0,"request_seq":12,"success":true,"command":"disconnect"}
```

**Reply by koinumasaoko** [reply]
This matter has been resolved. Thank you.
However, having made it to IBM i Debug v2.0.1, the Debug Service is giving me an error.
Is there anything that should be addressed or checked?

Attached is the error message after executing the following command
`JOBQ(QSYS/QUSRNOMAX) JOBD(QSYS/QSYSJOBD) USER(*CURRENT)`
> Debug Service job job 254242/KOINUMA/DBGSVCE submitted to job queue QUSRNOMAX in library QSYS. failed: Error: Data conversion or data mapping error. (22023).

The prerequisite PTFs for IBM i Debug have already been applied.
 I have also added the following to Debug Service.env.
`JAVA_HOME=/QOpenSys/QIBM/ProdData/JavaVM/jdk11/64bit`

**Reply by mkwan01** [reply, marked as answer]
The message "Data conversion or data mapping error." is not coming directly from the debugger. It may come from other dependent components. We can only do an investigation when a visible debug symptom is observed.

**Comment by cwbuck** [comment]
Would almost seem as though "Update production files" isn't being adhered too...

![image](https://github.com/user-attachments/assets/7c3bb5cf-c997-46ea-aa3c-77ddce8602af)

Debug Console:
EQARD1067I Remote Debug Service has generated the following trace files: 
/QIBM/UserData/IBMIDEBUGSERVICE/startDebugService_workspace/PDT_DAAS_20250214_143519_0460_1_DAP.txt
/QIBM/UserData/IBMIDEBUGSERVICE/startDebugService_workspace/PDT_DAAS_20250214_143519_0460_1_EPDC.hex
DBGP0003E  Program received unmonitored exception RNX1216: Error message CPF4203 appeared during OPEN for file VMLMSG1.

![image](https://github.com/user-attachments/assets/2a166db8-9544-4b1a-bff7-95aee2a7ae83)

**Reply by cwbuck** [reply]
I'm getting this message regardless of Update production files being turned on in my settings and can't seem to set UPDPROD(*YES) any other way

**Reply by worksofliam** [reply]
Is this during SEP or batch?

**Reply by cwbuck** [reply]
During SEP

**Comment by mkwan01** [comment]
@cwbuck Support for update production files in SEP debug requires IBM i debug extension v2.0.2 and the following host PTFs installed:
7.3 PTF SJ02996
7.4 PTF SJ03026
7.5 PTF SJ03030

Please verify whether you have updated the debug client and debug service host PTF. If you have right prerequisites, please check the "Debug trace" option and send us the output from the Debug Console view.
![image](https://github.com/user-attachments/assets/e084a4c3-68e0-45d2-bd3c-d4af364acbf8)
